modules/navigation/nav_mesh_generator_3d.cpp

```diff
@@ -388,29 +388,18 @@ void NavMeshGenerator3D::generator_parse_staticbody3d_node(const Ref<NavigationM
 
 							Vector<Vector3> vertex_array;
 							vertex_array.resize((heightmap_depth - 1) * (heightmap_width - 1) * 6);
-							int map_data_current_index = 0;
-
-							for (int d = 0; d < heightmap_depth; d++) {
-								for (int w = 0; w < heightmap_width; w++) {
-									if (map_data_current_index + 1 + heightmap_depth < map_data.size()) {
-										float top_left_height = map_data[map_data_current_index];
-										float top_right_height = map_data[map_data_current_index + 1];
-										float bottom_left_height = map_data[map_data_current_index + heightmap_depth];
-										float bottom_right_height = map_data[map_data_current_index + 1 + heightmap_depth];
-
-										Vector3 top_left = Vector3(start.x + w, top_left_height, start.y + d);
-										Vector3 top_right = Vector3(start.x + w + 1.0, top_right_height, start.y + d);
-										Vector3 bottom_left = Vector3(start.x + w, bottom_left_height, start.y + d + 1.0);
-										Vector3 bottom_right = Vector3(start.x + w + 1.0, bottom_right_height, start.y + d + 1.0);
-
-										vertex_array.push_back(top_right);
-										vertex_array.push_back(bottom_left);
-										vertex_array.push_back(top_left);
-										vertex_array.push_back(top_right);
-										vertex_array.push_back(bottom_right);
-										vertex_array.push_back(bottom_left);
-									}
-									map_data_current_index += 1;
+							Vector3 *vertex_array_ptrw = vertex_array.ptrw();
+							int vertex_index = 0;
+
+							for (int d = 0; d < heightmap_depth - 1; d++) {
+								for (int w = 0; w < heightmap_width - 1; w++) {
+									vertex_array_ptrw[vertex_index] = Vector3(start.x + w, map_data[(heightmap_width * d) + w], start.y + d);
+									vertex_array_ptrw[vertex_index + 1] = Vector3(start.x + w + 1, map_data[(heightmap_width * d) + w + 1], start.y + d);
+									vertex_array_ptrw[vertex_index + 2] = Vector3(start.x + w, map_data[(heightmap_width * d) + heightmap_width + w], start.y + d + 1);
+									vertex_array_ptrw[vertex_index + 3] = Vector3(start.x + w + 1, map_data[(heightmap_width * d) + w + 1], start.y + d);
+									vertex_array_ptrw[vertex_index + 4] = Vector3(start.x + w + 1, map_data[(heightmap_width * d) + heightmap_width + w + 1], start.y + d + 1);
+									vertex_array_ptrw[vertex_index + 5] = Vector3(start.x + w, map_data[(heightmap_width * d) + heightmap_width + w], start.y + d + 1);
```

Comment:
```suggestion
									vertex_array_ptrw[vertex_index] = start + Vector3(w, map_data[(heightmap_width * d) + w], d);
									vertex_array_ptrw[vertex_index + 1] = start + Vector3(w + 1, map_data[(heightmap_width * d) + w +1], d);
									vertex_array_ptrw[vertex_index + 2] = start + Vector3(w, map_data[(heightmap_width * d) + heightmap_width + w], d + 1);
									vertex_array_ptrw[vertex_index + 3] = start + Vector3(w + 1, map_data[(heightmap_width * d) + w +1], d);
									vertex_array_ptrw[vertex_index + 4] = start + Vector3(w + 1, map_data[(heightmap_width * d) + heightmap_width + w +1], d + 1);
									vertex_array_ptrw[vertex_index + 5] = start + Vector3(w, map_data[(heightmap_width * d) + heightmap_width + w], d + 1);
```
And make `start` a `Vector3`, for readability, same below.